qdrant-landing/content/articles/sparse-vectors.md

```diff
@@ -0,0 +1,391 @@
+
+# Leveraging Sparse Vectors in Qdrant for Efficient First-Stage Retrieval
+
+Qdrant supports a separate index for Sparse Vectors. This enables you to use the same collection for both dense and sparse vectors. Each "Point" in Qdrant can have both dense and sparse vectors.
+
+## Speed and Efficiency with Sparse Vectors
+
+Sparse vectors, inherently characterized by their minimalistic data representation, offer a significant advantage in terms of speed. 
+
+In Qdrant, the handling of sparse vectors is optimized to exploit this characteristic, resulting in rapid retrieval times. This makes Qdrant an ideal choice for first-stage retrieval where the goal is to quickly narrow down the search space from a large dataset. This is not a replacement for dense vectors, but a complementary approach that can be used in conjunction with dense vectors to improve ranking cost with minimal impact on relevance or latency. 
+
+## Practical Implementation in Python
+
+Let's delve into how Qdrant handles sparse vectors with an example. Here is what we will cover:
+
+1. Setting Up Qdrant Client: Initially, we establish a connection with Qdrant using the QdrantClient. This setup is crucial for subsequent operations.
+
+2. Creating a Collection with Sparse Vector Support: In Qdrant, a collection is a container for your vectors. Here, we create a collection specifically designed to support sparse vectors. This is done using the recreate_collection method where we define the parameters for sparse vectors, such as setting the index configuration.
+
+3. Inserting Sparse Vectors: Once the collection is set up, we can insert sparse vectors into it. This involves defining the sparse vector with its indices and values, and then upserting this point into the collection.
+
+4. Querying with Sparse Vectors: To perform a search, we first prepare a query vector. This involves computing the vector from a query text and extracting its indices and values. We then use these details to construct a query against our collection.
+
+5. Retrieving and Interpreting Results: The search operation returns results that include the id of the matching document, its score, and other relevant details. The score is a crucial aspect, reflecting the similarity between the query and the documents in the collection.
+
+### 1. Setting up
+
+```python
+# Qdrant client setup
+client = QdrantClient(":memory:")
+
+# Define collection name
+COLLECTION_NAME = "example_collection"
+
+# Insert sparse vector into Qdrant collection
+point_id = 1  # Assign a unique ID for the point
+```
+
+### 2. Creating a Collection with Sparse Vector Support
+
+```python
+client.recreate_collection(
+        collection_name=COLLECTION_NAME,
+        vectors_config={},
+        sparse_vectors_config={
+            "text": models.SparseVectorParams(
+                index=models.SparseIndexConfig(
+                    on_disk=False,
+                    full_scan_threshold=100,
```

Comment:
maybe we do not need to mention `full_scan_threshold` for now, it's a bit too deep

Reply:
Sounds good, I'll remove this